src/blocks/vpn.rs

```diff
@@ -0,0 +1,275 @@
+//!
+//! Placeholder | Value                                                     | Type   | Unit
+//! ------------|-----------------------------------------------------------|--------|------
+//! `icon`      | A static icon                                             | Icon   | -
+//! `country`   | Country currently connected to                            | Text   | -
+//! `flag`      | Country specific flag (depends on a font supporting them) | Text   | -
+//!
+//! Action    | Default button | Description
+//! ----------|----------------|-----------------------------------
+//! `toggle`  | Left           | toggles the vpn network connection
+//!
+//! # Drivers
+//!
+//! ## nordvpn
+//! Behind the scenes the nordvpn driver uses the `nordvpn` command line binary. In order for this to work
+//! properly the binary should be executable without root privileges.
+//!
+//! # Example
+//!
+//! Shows the current vpn network state:
+//!
+//! ```toml
+//! [[block]]
+//! block = "vpn"
+//! driver = "nordvpn"
+//! interval = 10
+//! format_connected = "VPN: $icon "
+//! format_disconnected = "VPN: $icon "
+//! state_connected = "good"
+//! state_disconnected = "warning"
+//! ```
+//!
+//! Possible values for `state_connected` and `state_disconnected`:
+//!
+//! ```
+//! Warning
+//! Critical
+//! Good
+//! Info
+//! Idle
+//! ```
+//!
+//! # Icons Used
+//!
+//! - `net_vpn`
+//! - `net_wired`
+//! - `net_down`
+//! - country code flags (if supported by font)
+//!
+//! Flags: They are not icons but unicode glyphs. You will need a font that
+//! includes them. Tested with: <https://www.babelstone.co.uk/Fonts/Flags.html>
+
+use regex::Regex;
+use std::process::Stdio;
+use tokio::process::Command;
+
+use crate::util::country_flag_from_iso_code;
+
+use super::prelude::*;
+
+#[derive(Deserialize, Debug, SmartDefault)]
+#[serde(rename_all = "snake_case")]
+enum DriverType {
+    #[default]
+    Nordvpn,
+}
+
+#[derive(Deserialize, Debug, SmartDefault)]
+#[serde(default)]
+pub struct Config {
+    driver: DriverType,
+    #[default(10.into())]
+    interval: Seconds,
+    format_connected: FormatConfig,
+    format_disconnected: FormatConfig,
+    state_connected: State,
+    state_disconnected: State,
+}
+
+enum Status {
+    Connected {
+        country: String,
+        country_flag: String,
+    },
+    Disconnected,
+    Error,
+}
+
+impl Status {
+    fn icon(&self) -> &str {
+        match self {
+            Status::Connected { .. } => "net_vpn",
+            Status::Disconnected => "net_wired",
+            Status::Error => "net_down",
+        }
+    }
+
+    fn flag(&self) -> &str {
+        match self {
+            Status::Connected { country_flag, .. } => country_flag.as_str(),
+            _ => "",
+        }
+    }
+
+    fn country(&self) -> &str {
+        match self {
+            Status::Connected { country, .. } => country.as_str(),
+            _ => "",
+        }
+    }
```

Comment:
These functions are not necessary anymore because you can just do 
```rust
Status::Connected { country, country_flag } => {
                widget.set_values(map!(
                        "icon" => Value::icon(api.get_icon(status.icon())?),
                        "country" => Value::text(country),
                        "flag" => Value::text(country_flag),
                ));
                widget.set_format(format_connected.clone());
                config.state_connected
            }
```

Reply:
Right!
I removed them.